test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
Not sure I have much to add beyond my previous comment :) If we're talking perfect world, I'd like to see

`set` -> `merge`
`assoc` -> `set`
`prop` -> `get`

but even if we're not at 1.0 yet having `prop` renamed to `get` might be a bit much to ask. So dropping the name `set` and un-deprecating/changing the current `merge` to this deep merge implementation of `set` seems reasonable to me.